syntax = "proto3";
import "google/protobuf/timestamp.proto";

message EmployeeCreateRequest {
  int32 biometricsId = 1;
  string name = 2;
  string rank = 3;
  optional EmployeeCreateRequestPhoto photo = 4;
}
message IdMessage {
  string id = 1;
}
message EmployeeReadResponse {
  string id = 1;
  int32 biometricsId = 2;
  string name = 3;
  string rank = 4;
  optional EmployeeReadResponsePhoto photo = 5;
}
message EmployeeUpdateRequest {
  string id = 1;
  int32 biometricsId = 2;
  string name = 3;
  string rank = 4;
  optional EmployeeUpdateRequestPhoto photo = 5;
}
message OkMessage {
}
message EmployeeListRequest {
  optional EmployeeFilters filters = 1;
  optional string search = 2;
  optional EmployeeSort sort = 3;
  optional EmployeeCursor cursor = 4;
  int32 limit = 5;
}
message EmployeeListResponse {
  repeated Employee items = 1;
  EmployeeCursor cursor = 2;
}
message WeaponCreateRequest {
  string name = 1;
  optional WeaponCreateRequestPhoto photo = 2;
}
message WeaponReadResponse {
  string id = 1;
  string name = 2;
  optional WeaponReadResponsePhoto photo = 3;
}
message WeaponUpdateRequest {
  string id = 1;
  string name = 2;
  optional WeaponUpdateRequestPhoto photo = 3;
}
message WeaponListRequest {
  optional WeaponFilters filters = 1;
  optional string search = 2;
  optional WeaponSort sort = 3;
  optional WeaponCursor cursor = 4;
  int32 limit = 5;
}
message WeaponListResponse {
  repeated Weapon items = 1;
  WeaponCursor cursor = 2;
}
message AmmoCreateRequest {
  string name = 1;
  optional AmmoCreateRequestPhoto photo = 2;
}
message AmmoReadResponse {
  string id = 1;
  string name = 2;
  optional AmmoReadResponsePhoto photo = 3;
}
message AmmoUpdateRequest {
  string id = 1;
  string name = 2;
  optional AmmoUpdateRequestPhoto photo = 3;
}
message AmmoListRequest {
  optional AmmoFilters filters = 1;
  optional string search = 2;
  optional AmmoSort sort = 3;
  optional AmmoCursor cursor = 4;
  int32 limit = 5;
}
message AmmoListResponse {
  repeated Ammo items = 1;
  AmmoCursor cursor = 2;
}
message AssignmentCreateRequest {
  AssignmentCreateRequestEmployee employee = 1;
  AssignmentCreateRequestWeapon weapon = 2;
  AssignmentCreateRequestAmmo ammo = 3;
  string dutyPost = 4;
  int32 ammoCount = 5;
}
message AssignmentReadResponse {
  string id = 1;
  AssignmentReadResponseEmployee employee = 2;
  AssignmentReadResponseWeapon weapon = 3;
  AssignmentReadResponseAmmo ammo = 4;
  string dutyPost = 5;
  int32 ammoCount = 6;
}
message AssignmentUpdateRequest {
  string id = 1;
  AssignmentUpdateRequestEmployee employee = 2;
  AssignmentUpdateRequestWeapon weapon = 3;
  AssignmentUpdateRequestAmmo ammo = 4;
  string dutyPost = 5;
  int32 ammoCount = 6;
}
message AssignmentListRequest {
  optional AssignmentFilters filters = 1;
  optional string search = 2;
  optional AssignmentSort sort = 3;
  optional AssignmentCursor cursor = 4;
  int32 limit = 5;
}
message AssignmentListResponse {
  repeated Assignment items = 1;
  AssignmentCursor cursor = 2;
}
message ScheduleAssignmentsAddRequest {
  string id = 1;
  AssignmentCreateRequest assignments = 2;
}
message ScheduleAssignmentsListRequest {
  optional AssignmentFilters filters = 1;
  optional string search = 2;
  optional AssignmentSort sort = 3;
  optional AssignmentCursor cursor = 4;
  int32 limit = 5;
  string parentId = 6;
}
message ScheduleAssignmentsListResponse {
  repeated Assignment items = 1;
  AssignmentCursor cursor = 2;
}
message ScheduleCreateRequest {
  google.protobuf.Timestamp startTime = 1;
  google.protobuf.Timestamp endTime = 2;
}
message ScheduleReadResponse {
  string id = 1;
  google.protobuf.Timestamp startTime = 2;
  google.protobuf.Timestamp endTime = 3;
  repeated ScheduleReadResponseAssignments assignments = 4;
}
message ScheduleUpdateRequest {
  string id = 1;
  google.protobuf.Timestamp startTime = 2;
  google.protobuf.Timestamp endTime = 3;
}
message ScheduleListRequest {
  optional ScheduleFilters filters = 1;
  optional string search = 2;
  optional ScheduleSort sort = 3;
  optional ScheduleCursor cursor = 4;
  int32 limit = 5;
}
message ScheduleListResponse {
  repeated Schedule items = 1;
  ScheduleCursor cursor = 2;
}
message CommonInfoCreateRequest {
  google.protobuf.Timestamp time = 1;
  optional CommonInfoCreateRequestAcknowledgedBy acknowledgedBy = 2;
  optional string remarks = 3;
}
message CommonInfoReadResponse {
  string id = 1;
  google.protobuf.Timestamp time = 2;
  optional CommonInfoReadResponseAcknowledgedBy acknowledgedBy = 3;
  optional string remarks = 4;
}
message CommonInfoUpdateRequest {
  string id = 1;
  google.protobuf.Timestamp time = 2;
  optional CommonInfoUpdateRequestAcknowledgedBy acknowledgedBy = 3;
  optional string remarks = 4;
}
message CommonInfoListRequest {
  optional CommonInfoFilters filters = 1;
  optional string search = 2;
  optional CommonInfoSort sort = 3;
  optional CommonInfoCursor cursor = 4;
  int32 limit = 5;
}
message CommonInfoListResponse {
  repeated CommonInfo items = 1;
  CommonInfoCursor cursor = 2;
}
message AmmoInfoCreateRequest {
  google.protobuf.Timestamp time = 1;
  optional AmmoInfoCreateRequestAcknowledgedBy acknowledgedBy = 2;
  optional string remarks = 3;
  int32 count = 4;
}
message AmmoInfoReadResponse {
  string id = 1;
  google.protobuf.Timestamp time = 2;
  optional AmmoInfoReadResponseAcknowledgedBy acknowledgedBy = 3;
  optional string remarks = 4;
  int32 count = 5;
}
message AmmoInfoUpdateRequest {
  string id = 1;
  google.protobuf.Timestamp time = 2;
  optional AmmoInfoUpdateRequestAcknowledgedBy acknowledgedBy = 3;
  optional string remarks = 4;
  int32 count = 5;
}
message AmmoInfoListRequest {
  optional AmmoInfoFilters filters = 1;
  optional string search = 2;
  optional AmmoInfoSort sort = 3;
  optional AmmoInfoCursor cursor = 4;
  int32 limit = 5;
}
message AmmoInfoListResponse {
  repeated AmmoInfo items = 1;
  AmmoInfoCursor cursor = 2;
}
message FlowsCreateRequest {
  FlowsCreateRequestAssignment assignment = 1;
  optional FlowsCreateRequestWeaponEntry weaponEntry = 2;
  optional FlowsCreateRequestClearingEntry clearingEntry = 3;
  optional FlowsCreateRequestAmmoEntry ammoEntry = 4;
  optional FlowsCreateRequestAmmoExit ammoExit = 5;
  optional FlowsCreateRequestClearingExit clearingExit = 6;
  optional FlowsCreateRequestWeaponExit weaponExit = 7;
}
message FlowsReadResponse {
  string id = 1;
  FlowsReadResponseAssignment assignment = 2;
  optional FlowsReadResponseWeaponEntry weaponEntry = 3;
  optional FlowsReadResponseClearingEntry clearingEntry = 4;
  optional FlowsReadResponseAmmoEntry ammoEntry = 5;
  optional FlowsReadResponseAmmoExit ammoExit = 6;
  optional FlowsReadResponseClearingExit clearingExit = 7;
  optional FlowsReadResponseWeaponExit weaponExit = 8;
}
message FlowsUpdateRequest {
  string id = 1;
  FlowsUpdateRequestAssignment assignment = 2;
  optional FlowsUpdateRequestWeaponEntry weaponEntry = 3;
  optional FlowsUpdateRequestClearingEntry clearingEntry = 4;
  optional FlowsUpdateRequestAmmoEntry ammoEntry = 5;
  optional FlowsUpdateRequestAmmoExit ammoExit = 6;
  optional FlowsUpdateRequestClearingExit clearingExit = 7;
  optional FlowsUpdateRequestWeaponExit weaponExit = 8;
}
message FlowsListRequest {
  optional FlowsFilters filters = 1;
  optional string search = 2;
  optional FlowsSort sort = 3;
  optional FlowsCursor cursor = 4;
  int32 limit = 5;
}
message FlowsListResponse {
  repeated Flows items = 1;
  FlowsCursor cursor = 2;
}
message FlowsOnScanRequest {
  int32 biometricsId = 1;
  Position position = 2;
}
message FlowsOnScanResponse {
  Position position = 1;
  bool red = 2;
  bool green = 3;
  bool buzzer = 4;
}
message AlertCreateRequest {
  AlertCreateRequestEmployee employee = 1;
  string alertType = 2;
  google.protobuf.Timestamp time = 3;
  optional string remarks = 4;
}
message AlertReadResponse {
  string id = 1;
  AlertReadResponseEmployee employee = 2;
  string alertType = 3;
  google.protobuf.Timestamp time = 4;
  optional string remarks = 5;
}
message AlertListRequest {
  optional AlertFilters filters = 1;
  optional string search = 2;
  optional AlertSort sort = 3;
  optional AlertCursor cursor = 4;
  int32 limit = 5;
}
message AlertListResponse {
  repeated Alert items = 1;
  AlertCursor cursor = 2;
}
message SignInRequest {
  string authName = 1;
  string password = 2;
}
message SignInResponse {
  string token = 1;
}
message FileObjectCreateRequest {
  string name = 1;
}
message FileObjectReadResponse {
  string id = 1;
  string name = 2;
}
message FileObjectUpdateRequest {
  string id = 1;
  string name = 2;
}
message FileObjectListRequest {
  optional FileObjectFilters filters = 1;
  optional string search = 2;
  optional FileObjectSort sort = 3;
  optional FileObjectCursor cursor = 4;
  int32 limit = 5;
}
message FileObjectListResponse {
  repeated FileObject items = 1;
  FileObjectCursor cursor = 2;
}
message EmployeeCreateRequestPhoto {
  string id = 1;
}
message EmployeeReadResponsePhoto {
  string id = 1;
  string name = 2;
}
message EmployeeUpdateRequestPhoto {
  string id = 1;
}
message EmployeeFilters {
  repeated string idIn = 1;
  optional int32 biometricsIdMin = 2;
  optional int32 biometricsIdMax = 3;
  optional string nameMin = 4;
  optional string nameMax = 5;
  optional string rankMin = 6;
  optional string rankMax = 7;
}
message EmployeeCursor {
  string id = 1;
  optional EmployeeFilters filters = 2;
  optional EmployeeSort sort = 3;
}
message Employee {
  string id = 1;
  int32 biometricsId = 2;
  string name = 3;
  string rank = 4;
  optional EmployeePhoto photo = 5;
}
message WeaponCreateRequestPhoto {
  string id = 1;
}
message WeaponReadResponsePhoto {
  string id = 1;
  string name = 2;
}
message WeaponUpdateRequestPhoto {
  string id = 1;
}
message WeaponFilters {
  repeated string idIn = 1;
  optional string nameMin = 2;
  optional string nameMax = 3;
}
message WeaponCursor {
  string id = 1;
  optional WeaponFilters filters = 2;
  optional WeaponSort sort = 3;
}
message Weapon {
  string id = 1;
  string name = 2;
  optional WeaponPhoto photo = 3;
}
message AmmoCreateRequestPhoto {
  string id = 1;
}
message AmmoReadResponsePhoto {
  string id = 1;
  string name = 2;
}
message AmmoUpdateRequestPhoto {
  string id = 1;
}
message AmmoFilters {
  repeated string idIn = 1;
  optional string nameMin = 2;
  optional string nameMax = 3;
}
message AmmoCursor {
  string id = 1;
  optional AmmoFilters filters = 2;
  optional AmmoSort sort = 3;
}
message Ammo {
  string id = 1;
  string name = 2;
  optional AmmoPhoto photo = 3;
}
message AssignmentCreateRequestEmployee {
  string id = 1;
}
message AssignmentCreateRequestWeapon {
  string id = 1;
}
message AssignmentCreateRequestAmmo {
  string id = 1;
}
message AssignmentReadResponseEmployee {
  string id = 1;
  string name = 2;
}
message AssignmentReadResponseWeapon {
  string id = 1;
  string name = 2;
}
message AssignmentReadResponseAmmo {
  string id = 1;
  string name = 2;
}
message AssignmentUpdateRequestEmployee {
  string id = 1;
}
message AssignmentUpdateRequestWeapon {
  string id = 1;
}
message AssignmentUpdateRequestAmmo {
  string id = 1;
}
message AssignmentFilters {
  repeated string idIn = 1;
  repeated string employeeIn = 2;
  repeated string weaponIn = 3;
  repeated string ammoIn = 4;
  optional string dutyPostMin = 5;
  optional string dutyPostMax = 6;
  optional int32 ammoCountMin = 7;
  optional int32 ammoCountMax = 8;
}
message AssignmentCursor {
  string id = 1;
  optional AssignmentFilters filters = 2;
  optional AssignmentSort sort = 3;
}
message Assignment {
  string id = 1;
  AssignmentEmployee employee = 2;
  AssignmentWeapon weapon = 3;
  AssignmentAmmo ammo = 4;
  string dutyPost = 5;
  int32 ammoCount = 6;
}
message ScheduleReadResponseAssignments {
  string id = 1;
  ScheduleReadResponseAssignmentsEmployee employee = 2;
  string dutyPost = 3;
}
message ScheduleFilters {
  repeated string idIn = 1;
  optional google.protobuf.Timestamp startTimeMin = 2;
  optional google.protobuf.Timestamp startTimeMax = 3;
  optional google.protobuf.Timestamp endTimeMin = 4;
  optional google.protobuf.Timestamp endTimeMax = 5;
  repeated string assignmentsIn = 6;
}
message ScheduleCursor {
  string id = 1;
  optional ScheduleFilters filters = 2;
  optional ScheduleSort sort = 3;
}
message Schedule {
  string id = 1;
  google.protobuf.Timestamp startTime = 2;
  google.protobuf.Timestamp endTime = 3;
}
message CommonInfoCreateRequestAcknowledgedBy {
  string id = 1;
}
message CommonInfoReadResponseAcknowledgedBy {
  string id = 1;
  string name = 2;
}
message CommonInfoUpdateRequestAcknowledgedBy {
  string id = 1;
}
message CommonInfoFilters {
  repeated string idIn = 1;
  optional google.protobuf.Timestamp timeMin = 2;
  optional google.protobuf.Timestamp timeMax = 3;
}
message CommonInfoCursor {
  string id = 1;
  optional CommonInfoFilters filters = 2;
  optional CommonInfoSort sort = 3;
}
message CommonInfo {
  string id = 1;
  google.protobuf.Timestamp time = 2;
  optional CommonInfoAcknowledgedBy acknowledgedBy = 3;
  optional string remarks = 4;
}
message AmmoInfoCreateRequestAcknowledgedBy {
  string id = 1;
}
message AmmoInfoReadResponseAcknowledgedBy {
  string id = 1;
  string name = 2;
}
message AmmoInfoUpdateRequestAcknowledgedBy {
  string id = 1;
}
message AmmoInfoFilters {
  repeated string idIn = 1;
  optional google.protobuf.Timestamp timeMin = 2;
  optional google.protobuf.Timestamp timeMax = 3;
  optional int32 countMin = 4;
  optional int32 countMax = 5;
}
message AmmoInfoCursor {
  string id = 1;
  optional AmmoInfoFilters filters = 2;
  optional AmmoInfoSort sort = 3;
}
message AmmoInfo {
  string id = 1;
  google.protobuf.Timestamp time = 2;
  optional AmmoInfoAcknowledgedBy acknowledgedBy = 3;
  optional string remarks = 4;
  int32 count = 5;
}
message FlowsCreateRequestAssignment {
  string id = 1;
}
message FlowsCreateRequestWeaponEntry {
  google.protobuf.Timestamp time = 1;
  optional FlowsCreateRequestWeaponEntryAcknowledgedBy acknowledgedBy = 2;
  optional string remarks = 3;
}
message FlowsCreateRequestClearingEntry {
  google.protobuf.Timestamp time = 1;
  optional FlowsCreateRequestClearingEntryAcknowledgedBy acknowledgedBy = 2;
  optional string remarks = 3;
}
message FlowsCreateRequestAmmoEntry {
  google.protobuf.Timestamp time = 1;
  optional FlowsCreateRequestAmmoEntryAcknowledgedBy acknowledgedBy = 2;
  optional string remarks = 3;
  int32 count = 4;
}
message FlowsCreateRequestAmmoExit {
  google.protobuf.Timestamp time = 1;
  optional FlowsCreateRequestAmmoExitAcknowledgedBy acknowledgedBy = 2;
  optional string remarks = 3;
  int32 count = 4;
}
message FlowsCreateRequestClearingExit {
  google.protobuf.Timestamp time = 1;
  optional FlowsCreateRequestClearingExitAcknowledgedBy acknowledgedBy = 2;
  optional string remarks = 3;
}
message FlowsCreateRequestWeaponExit {
  google.protobuf.Timestamp time = 1;
  optional FlowsCreateRequestWeaponExitAcknowledgedBy acknowledgedBy = 2;
  optional string remarks = 3;
}
message FlowsReadResponseAssignment {
  string id = 1;
  FlowsReadResponseAssignmentEmployee employee = 2;
  FlowsReadResponseAssignmentWeapon weapon = 3;
  FlowsReadResponseAssignmentAmmo ammo = 4;
  string dutyPost = 5;
  int32 ammoCount = 6;
}
message FlowsReadResponseWeaponEntry {
  string id = 1;
  google.protobuf.Timestamp time = 2;
  optional FlowsReadResponseWeaponEntryAcknowledgedBy acknowledgedBy = 3;
  optional string remarks = 4;
}
message FlowsReadResponseClearingEntry {
  string id = 1;
  google.protobuf.Timestamp time = 2;
  optional FlowsReadResponseClearingEntryAcknowledgedBy acknowledgedBy = 3;
  optional string remarks = 4;
}
message FlowsReadResponseAmmoEntry {
  string id = 1;
  google.protobuf.Timestamp time = 2;
  optional FlowsReadResponseAmmoEntryAcknowledgedBy acknowledgedBy = 3;
  optional string remarks = 4;
  int32 count = 5;
}
message FlowsReadResponseAmmoExit {
  string id = 1;
  google.protobuf.Timestamp time = 2;
  optional FlowsReadResponseAmmoExitAcknowledgedBy acknowledgedBy = 3;
  optional string remarks = 4;
  int32 count = 5;
}
message FlowsReadResponseClearingExit {
  string id = 1;
  google.protobuf.Timestamp time = 2;
  optional FlowsReadResponseClearingExitAcknowledgedBy acknowledgedBy = 3;
  optional string remarks = 4;
}
message FlowsReadResponseWeaponExit {
  string id = 1;
  google.protobuf.Timestamp time = 2;
  optional FlowsReadResponseWeaponExitAcknowledgedBy acknowledgedBy = 3;
  optional string remarks = 4;
}
message FlowsUpdateRequestAssignment {
  string id = 1;
}
message FlowsUpdateRequestWeaponEntry {
  string id = 1;
  google.protobuf.Timestamp time = 2;
  optional FlowsUpdateRequestWeaponEntryAcknowledgedBy acknowledgedBy = 3;
  optional string remarks = 4;
}
message FlowsUpdateRequestClearingEntry {
  string id = 1;
  google.protobuf.Timestamp time = 2;
  optional FlowsUpdateRequestClearingEntryAcknowledgedBy acknowledgedBy = 3;
  optional string remarks = 4;
}
message FlowsUpdateRequestAmmoEntry {
  string id = 1;
  google.protobuf.Timestamp time = 2;
  optional FlowsUpdateRequestAmmoEntryAcknowledgedBy acknowledgedBy = 3;
  optional string remarks = 4;
  int32 count = 5;
}
message FlowsUpdateRequestAmmoExit {
  string id = 1;
  google.protobuf.Timestamp time = 2;
  optional FlowsUpdateRequestAmmoExitAcknowledgedBy acknowledgedBy = 3;
  optional string remarks = 4;
  int32 count = 5;
}
message FlowsUpdateRequestClearingExit {
  string id = 1;
  google.protobuf.Timestamp time = 2;
  optional FlowsUpdateRequestClearingExitAcknowledgedBy acknowledgedBy = 3;
  optional string remarks = 4;
}
message FlowsUpdateRequestWeaponExit {
  string id = 1;
  google.protobuf.Timestamp time = 2;
  optional FlowsUpdateRequestWeaponExitAcknowledgedBy acknowledgedBy = 3;
  optional string remarks = 4;
}
message FlowsFilters {
  repeated string idIn = 1;
  repeated string assignmentIn = 2;
}
message FlowsCursor {
  string id = 1;
  optional FlowsFilters filters = 2;
  optional FlowsSort sort = 3;
}
message Flows {
  string id = 1;
  FlowsAssignment assignment = 2;
  optional FlowsWeaponEntry weaponEntry = 3;
  optional FlowsClearingEntry clearingEntry = 4;
  optional FlowsAmmoEntry ammoEntry = 5;
  optional FlowsAmmoExit ammoExit = 6;
  optional FlowsClearingExit clearingExit = 7;
  optional FlowsWeaponExit weaponExit = 8;
}
message AlertCreateRequestEmployee {
  string id = 1;
}
message AlertReadResponseEmployee {
  string id = 1;
  string name = 2;
}
message AlertFilters {
  repeated string idIn = 1;
  repeated string employeeIn = 2;
  optional string alertTypeMin = 3;
  optional string alertTypeMax = 4;
  optional google.protobuf.Timestamp timeMin = 5;
  optional google.protobuf.Timestamp timeMax = 6;
}
message AlertCursor {
  string id = 1;
  optional AlertFilters filters = 2;
  optional AlertSort sort = 3;
}
message Alert {
  string id = 1;
  AlertEmployee employee = 2;
  string alertType = 3;
  google.protobuf.Timestamp time = 4;
  optional string remarks = 5;
}
message FileObjectFilters {
  repeated string idIn = 1;
  optional string nameMin = 2;
  optional string nameMax = 3;
}
message FileObjectCursor {
  string id = 1;
  optional FileObjectFilters filters = 2;
  optional FileObjectSort sort = 3;
}
message FileObject {
  string id = 1;
  string name = 2;
}
message EmployeePhoto {
  string id = 1;
  string name = 2;
}
message WeaponPhoto {
  string id = 1;
  string name = 2;
}
message AmmoPhoto {
  string id = 1;
  string name = 2;
}
message AssignmentEmployee {
  string id = 1;
  string name = 2;
}
message AssignmentWeapon {
  string id = 1;
  string name = 2;
}
message AssignmentAmmo {
  string id = 1;
  string name = 2;
}
message ScheduleReadResponseAssignmentsEmployee {
  string id = 1;
  string name = 2;
}
message CommonInfoAcknowledgedBy {
  string id = 1;
  string name = 2;
}
message AmmoInfoAcknowledgedBy {
  string id = 1;
  string name = 2;
}
message FlowsCreateRequestWeaponEntryAcknowledgedBy {
  string id = 1;
}
message FlowsCreateRequestClearingEntryAcknowledgedBy {
  string id = 1;
}
message FlowsCreateRequestAmmoEntryAcknowledgedBy {
  string id = 1;
}
message FlowsCreateRequestAmmoExitAcknowledgedBy {
  string id = 1;
}
message FlowsCreateRequestClearingExitAcknowledgedBy {
  string id = 1;
}
message FlowsCreateRequestWeaponExitAcknowledgedBy {
  string id = 1;
}
message FlowsReadResponseAssignmentEmployee {
  string id = 1;
  string name = 2;
  optional FileObject photo = 3;
}
message FlowsReadResponseAssignmentWeapon {
  string id = 1;
  string name = 2;
  optional FileObject photo = 3;
}
message FlowsReadResponseAssignmentAmmo {
  string id = 1;
  string name = 2;
  optional FileObject photo = 3;
}
message FlowsReadResponseWeaponEntryAcknowledgedBy {
  string id = 1;
  string name = 2;
}
message FlowsReadResponseClearingEntryAcknowledgedBy {
  string id = 1;
  string name = 2;
}
message FlowsReadResponseAmmoEntryAcknowledgedBy {
  string id = 1;
  string name = 2;
}
message FlowsReadResponseAmmoExitAcknowledgedBy {
  string id = 1;
  string name = 2;
}
message FlowsReadResponseClearingExitAcknowledgedBy {
  string id = 1;
  string name = 2;
}
message FlowsReadResponseWeaponExitAcknowledgedBy {
  string id = 1;
  string name = 2;
}
message FlowsUpdateRequestWeaponEntryAcknowledgedBy {
  string id = 1;
}
message FlowsUpdateRequestClearingEntryAcknowledgedBy {
  string id = 1;
}
message FlowsUpdateRequestAmmoEntryAcknowledgedBy {
  string id = 1;
}
message FlowsUpdateRequestAmmoExitAcknowledgedBy {
  string id = 1;
}
message FlowsUpdateRequestClearingExitAcknowledgedBy {
  string id = 1;
}
message FlowsUpdateRequestWeaponExitAcknowledgedBy {
  string id = 1;
}
message FlowsAssignment {
  string id = 1;
}
message FlowsWeaponEntry {
  string id = 1;
  google.protobuf.Timestamp time = 2;
  optional FlowsWeaponEntryAcknowledgedBy acknowledgedBy = 3;
  optional string remarks = 4;
}
message FlowsClearingEntry {
  string id = 1;
  google.protobuf.Timestamp time = 2;
  optional FlowsClearingEntryAcknowledgedBy acknowledgedBy = 3;
  optional string remarks = 4;
}
message FlowsAmmoEntry {
  string id = 1;
  google.protobuf.Timestamp time = 2;
  optional FlowsAmmoEntryAcknowledgedBy acknowledgedBy = 3;
  optional string remarks = 4;
  int32 count = 5;
}
message FlowsAmmoExit {
  string id = 1;
  google.protobuf.Timestamp time = 2;
  optional FlowsAmmoExitAcknowledgedBy acknowledgedBy = 3;
  optional string remarks = 4;
  int32 count = 5;
}
message FlowsClearingExit {
  string id = 1;
  google.protobuf.Timestamp time = 2;
  optional FlowsClearingExitAcknowledgedBy acknowledgedBy = 3;
  optional string remarks = 4;
}
message FlowsWeaponExit {
  string id = 1;
  google.protobuf.Timestamp time = 2;
  optional FlowsWeaponExitAcknowledgedBy acknowledgedBy = 3;
  optional string remarks = 4;
}
message AlertEmployee {
  string id = 1;
  string name = 2;
}
message FlowsWeaponEntryAcknowledgedBy {
  string id = 1;
  string name = 2;
}
message FlowsClearingEntryAcknowledgedBy {
  string id = 1;
  string name = 2;
}
message FlowsAmmoEntryAcknowledgedBy {
  string id = 1;
  string name = 2;
}
message FlowsAmmoExitAcknowledgedBy {
  string id = 1;
  string name = 2;
}
message FlowsClearingExitAcknowledgedBy {
  string id = 1;
  string name = 2;
}
message FlowsWeaponExitAcknowledgedBy {
  string id = 1;
  string name = 2;
}
enum EmployeeSort {
  employeeSort_unspecified = 0;
  employeeSort_biometricsIdAsc = 1;
  employeeSort_biometricsIdDesc = 2;
  employeeSort_nameAsc = 3;
  employeeSort_nameDesc = 4;
  employeeSort_rankAsc = 5;
  employeeSort_rankDesc = 6;
}
enum WeaponSort {
  weaponSort_unspecified = 0;
  weaponSort_nameAsc = 1;
  weaponSort_nameDesc = 2;
}
enum AmmoSort {
  ammoSort_unspecified = 0;
  ammoSort_nameAsc = 1;
  ammoSort_nameDesc = 2;
}
enum AssignmentSort {
  assignmentSort_unspecified = 0;
  assignmentSort_dutyPostAsc = 1;
  assignmentSort_dutyPostDesc = 2;
  assignmentSort_ammoCountAsc = 3;
  assignmentSort_ammoCountDesc = 4;
}
enum ScheduleSort {
  scheduleSort_unspecified = 0;
  scheduleSort_startTimeAsc = 1;
  scheduleSort_startTimeDesc = 2;
  scheduleSort_endTimeAsc = 3;
  scheduleSort_endTimeDesc = 4;
}
enum CommonInfoSort {
  commonInfoSort_unspecified = 0;
  commonInfoSort_timeAsc = 1;
  commonInfoSort_timeDesc = 2;
}
enum AmmoInfoSort {
  ammoInfoSort_unspecified = 0;
  ammoInfoSort_timeAsc = 1;
  ammoInfoSort_timeDesc = 2;
  ammoInfoSort_countAsc = 3;
  ammoInfoSort_countDesc = 4;
}
enum FlowsSort {
  flowsSort_unspecified = 0;
}
enum Position {
  Position_unspecified = 0;
  Position_weapon = 1;
  Position_clearing = 2;
  Position_ammo = 3;
}
enum AlertSort {
  alertSort_unspecified = 0;
  alertSort_alertTypeAsc = 1;
  alertSort_alertTypeDesc = 2;
  alertSort_timeAsc = 3;
  alertSort_timeDesc = 4;
}
enum FileObjectSort {
  fileObjectSort_unspecified = 0;
  fileObjectSort_nameAsc = 1;
  fileObjectSort_nameDesc = 2;
}
